packages/api/cms-api/src/dam/files/files.resolver.ts

```diff
@@ -75,6 +86,25 @@ export function createFilesResolver({ File, Scope: PassedScope }: { File: Type<F
             return this.filesService.updateById(id, input);
         }
 
+        @Mutation(() => File)
+        @SkipBuild()
+        async importDamFileByDownload(
+            @Args("url", { type: () => String }) url: string,
+            @Args("scope", { type: () => Scope, defaultValue: hasNonEmptyScope ? undefined : {} }) scope: typeof Scope,
+            @Args("copyOfId", { type: () => ID }) copyOfId: string,
+            @Args("input", { type: () => UpdateFileInput }) { image: imageInput, ...input }: UpdateFileInput,
+        ): Promise<FileInterface> {
```

Comment:
Could this be a security risk? 

Since there is no validation of the url host, you can download files from any source. It could be possible to use this in an attack to make authorized users download a (malicious) file to the DAM.

Reply:
yes, you are right.

what needs to be checked?

- maximum file size
- mime type in an accepted mime type
- extension matched mime type
- svgContainsJavaScript 
- ...anything else?

Reply:
These four things are good. I can't think of anything else file-wise. 

But I think we might need to consider validating the hostname of the URL against a set of allowed options (the domains of the instances of a project). A file doesn't have to be malicious, it's enough if it's embarrassing or problematic. E.g. an image spreading false information or attacking a competitor hosted under the official domain of one of our clients. 
Don't you think that's a realistic risk?

We could disable this in dev mode. I don't think it's a huge limitation. Most projects have custom blocks that can't be imported into other projects anyway.

Reply:
I don't see any new risk here: if you can use the download api you can the upload api as well

Reply:
True

Reply:
implemented now